src/compiler/scala/tools/nsc/typechecker/Duplicators.scala

```diff
@@ -229,7 +229,12 @@ abstract class Duplicators extends Analyzer {
 
         case ddef @ DefDef(_, _, _, _, tpt, rhs) =>
           ddef.tpt modifyType fixType
-          super.typed(ddef.clearType(), mode, pt)
+          val result = super.typed(ddef.clearType(), mode, pt)
+          // TODO this is a hack, we really need a cleaner way to transport symbol attachments to duplicated methods
+          // bodies in specialized subclasses.
+          if (ddef.hasAttachment[DelambdafyTarget.type])
+            result.symbol.updateAttachment(DelambdafyTarget)
+          result
```

Comment:
I pasted the `ValDef` case because the `invalidSyms` connection is clearer there, sorry for the confusion. For `DefDef`s the symbol is thrown away and the type checker will create a new one...